test/algorithms/test_statepreparation.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2025 Chair for Design Automation, TUM
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * Licensed under the MIT License
+ */
+
+#include "algorithms/StatePreparation.hpp"
+#include "dd/Package.hpp"
+#include "dd/Simulation.hpp"
+#include "ir/QuantumComputation.hpp"
+
+#include <cmath>
+#include <complex>
+#include <cstddef>
+#include <gtest/gtest.h>
+#include <memory>
+#include <stdexcept>
+#include <utility>
```

Comment:
```suggestion
#include <tuple>
```
third time's the charm